site/src/Librecores/ProjectRepoBundle/Controller/ProjectController.php

```diff
@@ -250,14 +258,60 @@ public function settingsAction(Request $request, $parentName, $projectName)
         $form->handleRequest($request);
 
         if ($form->isSubmitted() && $form->isValid()) {
+            $projClassification = $request->request->get('classification');
+            $deleteClassification = $request->request->get('deleteClassification');
             $em = $this->getDoctrine()->getManager();
+            // Insert classifications
+            if (isset($projClassification)) {
+                foreach ($projClassification as $classification) {
+                    $projectClassification = new ProjectClassification();
+                    $projectClassification->setProject($p);
+                    $projectClassification->setClassification($classification);
+                    $projectClassification->setCreatedBy($p->getParentUser());
+                    $em->persist($projectClassification);
+                }
+            }
+            // Delete classifications
+            if (isset($deleteClassification)) {
+                foreach ($deleteClassification as $delete) {
+                    $projectClassification = $em->getRepository(ProjectClassification::class)->find($delete);
+                    $em->remove($projectClassification);
+                }
+            }
             $em->persist($p);
             $em->flush();
         }
 
+        // Retrive the project classifications for a project
+        $classifications = $this->getDoctrine()->getManager()
+            ->getRepository(ProjectClassification::class)
+            ->findBy(['project' => $p->getId()]);
+
+        // Retrive classification hierarchy
+        $classificationCategories = $this->getDoctrine()->getManager()
+            ->getRepository(ClassificationHierarchy::class)
+            ->findAll();
+
+        $classificationHierarchy = array();
+        $id = 0;
+        foreach ($classificationCategories as $category) {
+            $temp = array(
+                1 => $category->getId(),
+                2 => $category->getParent() == null ?
+                    $category->getParent(): $category->getParent()->getId(),
+                3 => $category->getName(),
```

Comment:
Why this strange numbering of the array starting with 1 (instead of 0)? Why don't you use named keys which would be more descriptive?